enumerables.rb

```diff
@@ -0,0 +1,243 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    h_array = to_a
+
+    if instance_of?(Range) == true
+      step do
+        yield(h_array[index])
+        index += 1
+      end
+    end
+    return self unless instance_of?(Range) == false
+
+    0.upto(length - 1) do |v|
+      yield(h_array[v])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    h_array = to_a
+    0.upto(length - 1) do |v|
+      yield(h_array[v], v)
+    end
+  end
```

Comment:
- **_`#my_each_with_index` when a block is given when self is an array returns the array itself after calling the given block once for each element in self_**
- **_`#my_each_with_index`, when a block is given when self is a range, returns the range itself after calling the given block once for each element in self_**
- **_`#my_each_with_index`, when a block is given when self is a range, behaves like `#each_with_index` calls the given block once for each element in self_**

Reply:
Hey, Temitayo. We just checked the my_each_with_index function and it does not return the self array. But the original function each_with_index does not return the self array either. Please check this.